### PR TITLE
Upgrade to latest versions of GitHub actions

### PR DESCRIPTION
**Pull request by electrocucaracha**
**What type of PR is this?**
dependency update

**Which issue does this PR fix?**:
NA

**What does this PR do / Why do we need it?**:
It seems like `dependabot` is not detecting the latest versions released by the GitHub actions used by this project. This change upgrades to the latest stable versions, resulting in fixing issues and vulneratibilties on the CI jobs.

**Testing done on this change**:
This testing has to be performed on the CI, which limits the possibility to run local tests.

**Will this PR introduce any new dependencies?**:
No

**Will this break upgrades or downgrades? Has updating a running cluster been tested?**:
No

**Does this change require updates to the CNI daemonset config files to work?**:
No

**Does this PR introduce any user-facing change?**:
No

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by orsenthil** [comment]
@electrocucaracha - Was this auto generated or manually updated? How can I verify these are latest? Also for ubuntu why do you suggest using the a version than main?

File: .github/workflows/issue-stale-pr.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
   stale:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/stale@main
+      - uses: actions/stale@v9
```
Comment (orsenthil) [inline .github/workflows/issue-stale-pr.yaml:15]: Why this is a versioned tag instead of main?

**Comment by electrocucaracha** [comment]
> @electrocucaracha - Was this auto generated or manually updated? How can I verify these are latest? Also for ubuntu why do you suggest using the a version than main? 

This was manually generated. Usually, these changes are detected by `dependabot`, but it seems not working as expected.

File: .github/workflows/issue-stale-pr.yaml (hunk 1 above)
Comment (electrocucaracha) [reply .github/workflows/issue-stale-pr.yaml:15]: It's a good practice to not use the main branch, so I suggest using the latest stable

**Comment by orsenthil** [comment]
@electrocucaracha - would you be interested in pinning the commit hash instead of the tags of these packages; there is a reference to use commit hash here.

**Comment by electrocucaracha** [comment]
> @electrocucaracha - would you be interested in pinning the commit hash instead of the tags of these packages; there is a reference to use commit hash here.

Ok, I can give it a try. Maybe this can be achieved configuring the `dependabot` [configuration](https://github.com/aws/amazon-vpc-cni-k8s/blob/master/.github/dependabot.yml), properly, for now it requires adding this section:

```
  - package-ecosystem: "github-actions"
    directory: "/"
    schedule:
      interval: "monthly"
```

**Comment by orsenthil** [comment]
Yes, we can add that section in the configuration too.

**Comment by electrocucaracha** [comment]
@orsenthil Apparently `dependabot` doesn't support commit hashes feature, and some [third party tools](https://blog.rafaelgss.dev/why-you-should-pin-actions-by-commit-hash) offer that lack of functionality. But this can be easily created with the following bash script.

```bash
gh_actions=$(grep -r "uses: [a-z\-]*/[a-z\-]*@" .github/workflows/ | sed 's/@.*//' | awk -F ': ' '{ print $3 }' | sort | uniq)

for action in $gh_actions; do
    commit_hash=$(git ls-remote --tags "https://github.com/$action" | awk '{ print $NF,$0 }' | sort -k1,1 -n | cut -f2- -d' ' | grep -oh '.*refs/tags/v[0-9]*$' | tail -1 | awk '{ printf "%s # %s\n",$1,$2 }')
    egrep -lRZ "uses: $action@" .github/workflows/ | xargs -0 -l sed -i -e "s|uses: $action@.*|uses: $action@$commit_hash|g"
done
```

Maybe, those bash scripts instructions can be included in a GitHub action.

**Comment by orsenthil** [comment]
@electrocucaracha If the bash scripts help update to the latest action commit sha, we can include them as a makefile target as a first step. We want to control on when to update

**Comment by electrocucaracha** [comment]
> @electrocucaracha If the bash scripts help update to the latest action commit sha, we can include them as a makefile target as a first step. We want to control on when to update

Well, I had in mind to use it in a GitHub action that uses this https://github.com/marketplace/actions/create-pr-action

**Comment by orsenthil** [comment]
>  I had in mind to use it in a GitHub action that uses this https://github.com/marketplace/actions/create-pr-action

Sounds good to me.

**Review by orsenthil — APPROVED** [review]

File: .github/workflows/update.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+name: Scheduled Update Versions
+# yamllint disable-line rule:truthy
+on:
+  schedule:
+    - cron: '0 0 * * 5'
+  workflow_dispatch:
+jobs:
+  check-versions:
+    permissions:
+      contents: write  # for technote-space/create-pr-action to push code
```
Comment (orsenthil) [inline .github/workflows/update.yml:19]: I am wondering if write action is required here?

**Review by orsenthil — CHANGES_REQUESTED** [review]
Is write to repo authorization required for the GH action?

File: .github/workflows/update.yml (hunk 2 above)
Comment (electrocucaracha) [reply .github/workflows/update.yml:19]: I think that was required for [protected tags](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/configuring-tag-protection-rules), I'll remove it to confirm that.

**Review by orsenthil — APPROVED** [review]
LGTM.

**Pull request merged**